### PR TITLE
Add documentation for MapStyle editor

### PR DESCRIPTION
**Pull request by matteblair**

### CODE REVIEW

File: Documentation/MapStyleEditor.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+![MapStyle editor interface](Images/MapStyleEditor.png)
+
+You can edit and save the settings for your map in an asset called a Map Style. To create a new Map Style, go to the 'Assets' menu and choose 'Create' -> 'Mapzen' -> 'MapStyle'. To edit the Map Style, select it from the project window and then use the inspector window to view and change settings.
+
+Editing Options
+---------------
+
+![MapStyle editing options](Images/MapStyleEditingOptions.png)
+
+If you enable the option to update while editing, any RegionMap instances in the current scene that use this MapStyle will be rebuilt using the current MapStyle settings whenever a change is made. **IMPORTANT:** These automatic updates will overwrite any manual edits you have made to the map objects in the scene.
```
Comment (tallytalwar) [inline Documentation/MapStyleEditor.md]: Nitpick.. probably move this **IMPORTANT** to next line.

File: Documentation/MapStyleEditor.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+-------|------------------------------------------
+All Of | match all of the Matchers in the list
+Any Of | match any of the Matchers in the list
+None Of | match any of the Matchers in the list
+
+### Style
+
+Each of the features in this Layer will be turned into renderable meshes using the builders enabled in the Style section.
+
+### Polygon Builder
```
Comment (tallytalwar) [inline Documentation/MapStyleEditor.md]: Probably make sense to mention the default values here.

Comment (tallytalwar) [reply Documentation/MapStyleEditor.md]: Example in `extrusion` or `UV Mode` properties.

Comment (matteblair) [reply Documentation/MapStyleEditor.md]: Actually I think it's simplest to not name the default values. Unlike in a Tangram scene file, values can't be omitted. So the current value of a field is always present and visible, whether it is default or not.

**Pull request merged**